### PR TITLE
feat: Add SSL/TLS support

### PR DESCRIPTION
**Pull request by stsnel**
This makes it possible to use the iRODS storage plugin to connect to iRODS environments that enforce use of SSL/TLS

### CODE REVIEW

File: snakemake_storage_plugin_irods/__init__.py [hunk 1]
```diff
@@ -94,6 +94,84 @@ class StorageProviderSettings(StorageProviderSettingsBase):
...
+    encryption_salt_size: int = field(
+        default=8,
+        metadata={
+            "help": f"Salt size for parallel transfer encryption. {env_msg}",
+            "env_var": False,
+            "required": False,
+        },
+    )
+    client_server_negotiation: str = field(
+        default="",
```
Comment (johanneskoester) [inline snakemake_storage_plugin_irods/__init__.py]:
```suggestion
    client_server_negotiation: Optional[str] = field(
        default=None,
```

File: snakemake_storage_plugin_irods/__init__.py [hunk 2]
```diff
@@ -94,6 +94,84 @@ class StorageProviderSettings(StorageProviderSettingsBase):
...
+            + " validity (but not hostname), "
+            + f"hostname: verify certificate validity and hostname. {env_msg}",
+            "env_var": False,
+            "required": False,
+        },
+    )
+    ssl_ca_certificate_file: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "File with trusted CA certificates in PEM format. "
```
Comment (johanneskoester) [inline snakemake_storage_plugin_irods/__init__.py]:
```suggestion
            "help": "Path to file with trusted CA certificates in PEM format. "
```

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Thanks a lot! Some minor things below.

**Comment by stsnel** [comment]
Thanks! I've included the suggested changes in the commit

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**